### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will update the doxygen generation files and fix some warnings and issue with the C API,  clear up some remaining deprecated items, and rename some of the federate_id related files.

### CODE REVIEW

**Review by nightlark — COMMENTED** [review]
It looks like some key words (`QUERY`, `FILTER`, `INPUT`, etc) were changed to all caps regardless of where they appear.

**Comment by phlptp** [comment]
I think I fixed those,  I must have missed a box for the automatic renaming in visual studio.

File: docs/Public_API.md [hunk 1]
```diff
@@ -62,14 +52,13 @@ The public API includes the following
...
   - Tracer.hpp
-  - Clone.hpp (new in 2.2)
+  - Clone.hpp
   - helicsApp.hpp
   - BrokerApp.hpp (aliased to application_api version)
   - CoreApp.hpp (aliased to application_api version)
-  - BrokerServer.hpp (removed in 2.3 as not useful for library operations, though still available in the static library)
 
-- Exceptions: Any function dealing with Inputs concerning data from multiple sources is subject to change, the vector subscription Objects, and vector Input objects are subject to change. Also some changes may occur in regard to units on the Application API.
+- Exceptions: The vector subscription Objects, and vector Input objects are subject to change.
```
Comment (nightlark) [inline docs/Public_API.md:60]:
```suggestion
- Exceptions: The vector subscription Objects, and vector Input objects are subject to change.
```

File: src/docs/c_api_reference.dox [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /**
  \page C_api_index C API function reference
 \section enumerations Helics Enumerations s
- Many of the functions take properties or options that are defined by the enumerations available in the HELICS library see \ref helics_enums.h
+ Many of the functions take properties or options that are defined by the enumerations available in the HELICS library see \ref helics_enums.h
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:4]:
```suggestion
 Many of the functions take properties or options that are defined by the enumerations available in the HELICS library see \ref helics_enums.h
```

File: src/docs/c_api_reference.dox [hunk 3]
```diff
@@ -38,8 +38,8 @@ Index of functions available in the C and by extension the language API's
  - \ref helicsCreateFederateInfo
  - \ref helicsCreateQuery
 
- \section core Core Functions
-functions applying to a \ref helics_core object
+ \section core Core Functions
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:41]:
```suggestion
 \section core Core Functions
```

File: src/docs/c_api_reference.dox [hunk 4]
```diff
@@ -63,8 +63,8 @@ functions applying to a \ref helics_core object
  - \ref helicsCoreRegisterFilter
  - \ref helicsCoreRegisterCloningFilter
 
- \section broker Broker Functions
-Functions applying to a \ref helics_broker object
+ \section broker Broker Functions
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:66]:
```suggestion
 \section broker Broker Functions
```

File: src/docs/c_api_reference.dox [hunk 5]
```diff
@@ -86,8 +86,8 @@ Functions applying to a \ref helics_broker object
  - \ref helicsBrokerGlobalError
  - \ref helicsBrokerSetLoggingCallback
 
- \section fedinfo Federate Info Functions
-Functions applying to a \ref helics_federate_info object
+ \section fedinfo Federate Info Functions
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:89]:
```suggestion
 \section fedinfo Federate Info Functions
```

File: src/docs/c_api_reference.dox [hunk 6]
```diff
@@ -164,8 +164,8 @@ Functions applying to all \ref helics_federate objects
  - \ref helicsFederateSetLoggingCallback
  - \ref helicsFederateSetQueryCallback
 
- \section vfederate Value Federate Functions
-functions applying to federates created as a value or combination federate \ref helics_federate objects
+ \section vfederate Value Federate Functions
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:167]:
```suggestion
 \section vfederate Value Federate Functions
```

File: src/docs/c_api_reference.dox [hunk 7]
```diff
@@ -105,8 +105,8 @@ Functions applying to a \ref helics_federate_info object
  - \ref helicsFederateInfoSetTimeProperty
  - \ref helicsFederateInfoSetIntegerProperty
 
- \section federate Federate Functions
-Functions applying to all \ref helics_federate objects
+ \section federate Federate Functions
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:108]:
```suggestion
 \section federate Federate Functions
```

File: src/docs/c_api_reference.dox [hunk 8]
```diff
@@ -257,23 +257,23 @@ functions applying to a \ref helics_input object
  - \ref helicsInputLastUpdateTime
  - \ref helicsInputClearUpdate
 
- \section message Message Federate Functions
-Functions applying to federates created as a value or combination federate \ref helics_federate objects
+ \section message Message Federate Functions
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:260]:
```suggestion
 \section message Message Federate Functions
```

File: src/docs/c_api_reference.dox [hunk 9]
```diff
@@ -211,8 +211,8 @@ functions applying to a \ref helics_publication object
  - \ref helicsPublicationSetOption
  - \ref helicsPublicationSetMinimumChange
 
- \section input Input interface Functions
-functions applying to a \ref helics_input object
+ \section input Input interface Functions
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:214]:
```suggestion
 \section input Input interface Functions
```

File: src/docs/c_api_reference.dox [hunk 10]
```diff
@@ -303,8 +303,8 @@ functions applying to a \ref helics_message_object
  - \ref helicsMessageFree
  - \ref helicsMessageClear
 
- \section endpoint Endpoint interface Functions
-functions applying to a \ref helics_endpoint object
+ \section endpoint Endpoint interface Functions
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:306]:
```suggestion
 \section endpoint Endpoint interface Functions
```

File: src/docs/c_api_reference.dox [hunk 11]
```diff
@@ -257,23 +257,23 @@ functions applying to a \ref helics_input object
...
- - \ref helicsFederatePendingMessagesCount
+ - \ref helicsFederatePendingMessageCount
  - \ref helicsFederateGetMessage
  - \ref helicsFederateCreateMessage
  - \ref helicsFederateClearMessages
  - \ref helicsFederateGetEndpointCount
 
- \section messageObject Message object Functions
-functions applying to a \ref helics_message_object
+ \section messageObject Message object Functions
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:275]:
```suggestion
 \section messageObject Message object Functions
```

File: src/docs/c_api_reference.dox [hunk 12]
```diff
@@ -343,8 +343,8 @@ Functions applying to all federates but related to \ref helics_filter
  - \ref helicsFederateGetFilter
  - \ref helicsFederateGetFilterByIndex
 
- \section filter Filter interface Functions
-Functions applying to a \ref helics_filter object
+ \section filter Filter interface Functions
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:346]:
```suggestion
 \section filter Filter interface Functions
```

File: src/docs/c_api_reference.dox [hunk 13]
```diff
@@ -333,8 +333,8 @@ functions applying to a \ref helics_endpoint object
  - \ref helicsEndpointAddSourceFilter
  - \ref helicsEndpointAddDestinationFilter
 
- \section filterfed Filter Federate Functions
-Functions applying to all federates but related to \ref helics_filter
+ \section filterfed Filter Federate Functions 
+Functions applying to all federates but related to \ref HelicsFilter
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:337]:
```suggestion
 \section filterfed Filter Federate Functions
Functions applying to all federates but related to \ref HelicsFilter
```

File: src/docs/c_api_reference.dox [hunk 14]
```diff
@@ -362,8 +362,8 @@ Functions applying to a \ref helics_filter object
  - \ref helicsFilterSetOption
  - \ref helicsFilterGetOption
 
- \section query Query Functions
-functions applying to a \ref helics_query object
+ \section query Query Functions
```
Comment (nightlark) [inline src/docs/c_api_reference.dox:365]:
```suggestion
 \section query Query Functions
```

File: src/helics/application_api/Inputs.hpp [hunk 15]
```diff
@@ -178,8 +178,8 @@ class HELICS_CXX_EXPORT Input: public Interface {
 
     /** register a callback for an update notification
     @details the callback is called in the just before the time request function returns
-    @param callback a function with signature void( Time time)
-    time is the time the value was updated  This callback is a notification callback and doesn't
+    @param callback a function with signature void( Time time) time is the time the value was updated.
```
Comment (nightlark) [inline src/helics/application_api/Inputs.hpp:181]:
```suggestion
    @param callback a function with signature void(Time time) where time is the time the value was updated.
```

File: src/helics/core/Core.hpp [hunk 16]
```diff
@@ -540,6 +531,7 @@ class Core {
     filter
     @param handle the identifier of the interface
     @param name the name of the filter or its target
+    @param hint the interface type for the destination target
```
Comment (nightlark) [inline src/helics/core/Core.hpp:534]:
```suggestion
    @param hint the interface type for the source target
```

File: src/helics/shared_api_library/MessageFederate.h [hunk 17]
```diff
@@ -425,7 +414,7 @@ HELICS_EXPORT void helicsEndpointSetInfo(HelicsEndpoint endpoint, const char* in
  * Get the data in a specified tag of an endpoint
  *
  * @param endpoint The endpoint to query.
- *
+ * @param tagname the name of the tag to query
```
Comment (nightlark) [inline src/helics/shared_api_library/MessageFederate.h:417]:
```suggestion
 * @param tagname The name of the tag to query.
```

File: src/helics/shared_api_library/MessageFederate.h [hunk 18]
```diff
@@ -511,10 +500,9 @@ HELICS_EXPORT void helicsEndpointAddSourceFilter(HelicsEndpoint endpoint, const
  * add a destination filter to an endpoint
  *
  * @param endpoint The endpoint to modify.
- * @param targetEndpoint the name of the filter to add
+ * @param filterName the name of the filter to add
```
Comment (nightlark) [inline src/helics/shared_api_library/MessageFederate.h:503]:
```suggestion
 * @param filterName The name of the filter to add.
```

File: src/helics/shared_api_library/MessageFilters.h [hunk 19]
```diff
@@ -288,15 +288,15 @@ HELICS_EXPORT void helicsFilterSetInfo(HelicsFilter filt, const char* info, Heli
  * Get the data in a specified tag of a filter.
  *
  * @param filt The filter to query.
- *
+ * @param tagname the name of the tag to query
```
Comment (nightlark) [inline src/helics/shared_api_library/MessageFilters.h:291]:
```suggestion
 * @param tagname The name of the tag to query.
```

File: src/helics/shared_api_library/ValueFederate.h [hunk 20]
```diff
@@ -885,15 +842,14 @@ HELICS_EXPORT const char* helicsInputGetInfo(HelicsInput inp);
...
-
  */
 HELICS_EXPORT void helicsInputSetInfo(HelicsInput inp, const char* info, HelicsError* err);
 
 /**
  * Get the data in a specified tag of an input.
  *
  * @param inp The input object to query.
- *
+ * @param tagname the name of the tag to get the value for.
```
Comment (nightlark) [inline src/helics/shared_api_library/ValueFederate.h:852]:
```suggestion
 * @param tagname The name of the tag to get the value for.
```

File: src/helics/shared_api_library/ValueFederate.h [hunk 21]
```diff
@@ -926,28 +881,26 @@ HELICS_EXPORT const char* helicsPublicationGetInfo(HelicsPublication pub);
...
-
  */
 HELICS_EXPORT void helicsPublicationSetInfo(HelicsPublication pub, const char* info, HelicsError* err);
 
 /**
  * Get the data in a specified tag of a publication.
  *
  * @param pub The publication object to query.
- *
+ * @param tagname the name of the tag to query.
```
Comment (nightlark) [inline src/helics/shared_api_library/ValueFederate.h:891]:
```suggestion
 * @param tagname The name of the tag to query.
```

File: src/helics/shared_api_library/ValueFederate.h [hunk 22]
```diff
@@ -926,28 +881,26 @@ HELICS_EXPORT const char* helicsPublicationGetInfo(HelicsPublication pub);
...
  */
 HELICS_EXPORT const char* helicsPublicationGetTag(HelicsPublication pub, const char* tagname);
 
 /**
  * Set the data in a specific tag for a publication.
  *
- * @param pub The publication object to query.
- * @param tagname The string to set.
+ * @param pub The publication object to set a tag for.
+ * @param tagname The name of string to set.
```
Comment (nightlark) [inline src/helics/shared_api_library/ValueFederate.h:900]:
```suggestion
 * @param tagname The name of the tag to set.
```

**Review by nightlark — APPROVED** [review]

**Pull request merged**